### ISSUE TITLE
sphinx.build_main and sphinx.make_main deprecated, but throw NameError

### ISSUE THREAD

**Issue by ViktorHaag** [sphinx-doc/sphinx#4669]
Subject: Attempts to directly invoke sphinx.build_main() or sphinx.make_main() result in NameError

### Problem
- I have a wrapper script that directly invokes sphinx.make_main(); with Sphinx 1.7.1 this now results in a NameError exception instead of just the intended deprecation warning.

#### Procedure to reproduce the problem

Attempt to directly call the (now deprecated) sphinx.make_main().

#### Error logs / results
Here's the relevant traceback:
```
Traceback (most recent call last):
  File "./build-deploy.py", line 141, in <module>
    errno = build_html(args.sourcedir, args.outputdir, args.branch, args.nitpicky, args.force)
  File "./build-deploy.py", line 65, in build_html
    errno = _sphinx_build(arglist)
  File "./build-deploy.py", line 44, in _sphinx_build
    return sphinx.make_main(arglist)
  File "/usr/local/lib/python3.6/site-packages/sphinx/__init__.py", line 91, in make_main
    return build.make_main(argv[1:])  # skip first argument to adjust arguments (refs: #4615)
NameError: name 'build' is not defined
```

#### Expected results
I'd expect to get the deprecation warning, but for the invocation to still work, as it would with sphinx.main() where I see:
```
def main(*args, **kwargs):
    from .cmd import build
```

I realize that I'm likely calling into a function that's not part of the Sphinx public API, but I don't see much reason for the warnings.warn() calls to be there in addition to the exception. It might be friendlier if both sphinx.build_main() and sphinx.make_main() also did the import of build module so they continued to work as well as warn.

### Environment info
- OS: MacOS 10.12.6
- Python version: Python 3.6.4
- Sphinx version: 1.7.1

**Comment by ViktorHaag** [comment]
Note, I have migrated my wrapper code to invoke from sphinx.cmd.build.make_main(), and the move of the entry point seems fine to me -- I only think that this change between 1.7.0 and 1.7.1 should warn but necessarily break. If maintainers feel this is unwarranted and want to simply close this issue, that's fine.

Related issues: #4615 and #4623

**Comment by tk0miya** [comment]
OMG, that is not intended...

**Comment by tk0miya** [comment]
Fixed by #4672.
Thank you for reporting
